tests/ops/op_math/test_adjoint_op.py

```diff
@@ -0,0 +1,599 @@
+        base_eigvals = base.eigvals()
+        adj_eigvals = Adjoint(base).eigvals()
+        assert qml.math.allclose(base_eigvals, adj_eigvals)
+
+    def test_non_hermitian_eigvals(self):
+        """Test that the Adjoint eigvals are the conjugate of the base's eigvals."""
+
+        base = qml.SX(0)
+        base_eigvals = base.eigvals()
+        adj_eigvals = Adjoint(base).eigvals()
+
+        assert qml.math.allclose(qml.math.conj(base_eigvals), adj_eigvals)
+
+    def test_no_matrix_defined_eigvals(self):
+        """Test that if the base does not define eigvals, The Adjoint raises the same error."""
+        base = qml.QubitStateVector([1, 0], wires=0)
+
+        with pytest.raises(qml.operation.EigvalsUndefinedError):
+            Adjoint(base).eigvals()
+
+
+class TestDecompositionExpand:
+    """Test the decomposition and expand methods for the Adjoint class."""
+
+    def test_decomp_custom_adjoint_defined(self):
+        """Test decomposition method when a custom adjoint is defined."""
+        decomp = Adjoint(qml.Hadamard(0)).decomposition()
+        assert len(decomp) == 1
+        assert isinstance(decomp[0], qml.Hadamard)
+
+    def test_expand_custom_adjoint_defined(self):
+        """Test expansion method when a custom adjoint is defined."""
+        base = qml.Hadamard(0)
+        tape = Adjoint(base).expand()
+
+        assert len(tape) == 1
+        assert isinstance(tape[0], qml.Hadamard)
+
+    def test_decomp(self):
+        """Test decomposition when base has decomposition but no custom adjoint."""
+        base = qml.SX(0)
+        base_decomp = base.decomposition()
+        decomp = Adjoint(base).decomposition()
+
+        for adj_op, base_op in zip(decomp, reversed(base_decomp)):
+            assert isinstance(adj_op, Adjoint)
+            assert adj_op.base.__class__ == base_op.__class__
+            assert qml.math.allclose(adj_op.data, base_op.data)
+
+    def test_expand(self):
+        """Test expansion when base has decomposition but no custom adjoint."""
+
+        base = qml.SX(0)
+        base_tape = base.expand()
+        tape = Adjoint(base).expand()
+
+        for base_op, adj_op in zip(reversed(base_tape), tape):
+            assert isinstance(adj_op, Adjoint)
+            assert base_op.__class__ == adj_op.base.__class__
+            assert qml.math.allclose(adj_op.data, base_op.data)
+
+    def test_no_base_gate_decomposition(self):
+        """Test that when the base gate doesn't have a decomposition, the Adjoint decomposition
+        method raises the proper error."""
+        nr_wires = 2
+        rho = np.zeros((2**nr_wires, 2**nr_wires), dtype=np.complex128)
+        rho[0, 0] = 1  # initialize the pure state density matrix for the |0><0| state
+        base = qml.QubitDensityMatrix(rho, wires=(0, 1))
+
+        with pytest.raises(qml.operation.DecompositionUndefinedError):
+            Adjoint(base).decomposition()
+
+
+class TestIntegration:
```

Comment:
I would imagine we already have tests (not sure where) which pass an `Adjoint(op)` into a measurement process and assert the correct result, but it might be good to double check that or included it here.

Reply:
I'd need to explicitly add `Adjoint(PauliX)`, etc. to the list of `obersvables`.  We can do that later.